lib/redbird/key.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Redbird.Key do
+  alias Redbird.Crypto
+
+  def generate(namespace \\ namespace()) do
+    namespace <> (96 |> :crypto.strong_rand_bytes() |> Base.encode64())
+  end
+
+  def sign_key(key, conn) do
+    {:ok, key, namespace} = extract_key(key)
+    to_string(namespace) <> Crypto.sign_key(key, conn)
+  end
+
+  def verify(key, conn) do
+    Crypto.verify_key(key, conn)
+  end
+
+  def deletable?(key, conn) do
+    with {:ok, key, _} <- extract_key(key),
+         {:ok, _verified_key} <- Crypto.verify_key(key, conn) do
+      true
+    else
+      _ -> false
+    end
+  end
+
+  def extract_key(candidate) when is_binary(candidate) do
```

Comment:
Do we need this to be a public function? Aside from tests, it seems like we only use it within this module. What do you think? Would it be easy to test this behavior through the public function or is this helpful to have to easily extract the key for other tests (i.e. not the test testing `extract_key/1`)?

Reply:
My tendency is to leave it a public function. You're right that it's not called outside of tests but it's also a pure function that operates on immutable data so I don't see a drawback to it being public and can see potential benefit to being able to dissect the Redis key into its constituent Redbird parts. Do you see downsides to it being public?

Reply:
I don't see downsides yet, but I try to default things to private functions unless they need to be public. And I do that because it allows me to change them without having to worry about who's using them. In this case, the project isn't large and it's probably fine. So I would go with your gut. And if it's helpful, even if only from other tests, then maybe it makes sense for it to be public.